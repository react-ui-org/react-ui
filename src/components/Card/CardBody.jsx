import PropTypes from 'prop-types';
import React from 'react';
import { withGlobalProps } from '../../providers/globalProps';
import { transferProps } from '../../jsHelpers/transferProps/transferProps';
import styles from './Card.module.scss';

export const CardBody = ({
  children,
  ...restProps
}) => (
  <div
    {...transferProps(restProps)}
    className={styles.body}
  >
    {children}
  </div>
);

CardBody.propTypes = {
  /**
   * Content of the card.
   */
  children: PropTypes.node.isRequired,
};

export const CardBodyWithGlobalProps = withGlobalProps(CardBody, 'CardBody');

export default CardBodyWithGlobalProps;
