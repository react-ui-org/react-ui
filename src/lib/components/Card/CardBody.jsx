import PropTypes from 'prop-types';
import React from 'react';
import { withProviderContext } from '../../provider';
import styles from './Card.scss';

export const CardBody = (props) => {
  const {
    children,
  } = props;

  return (
    <div className={styles.body}>
      {children}
    </div>
  );
};

CardBody.propTypes = {
  /**
   * Content of the card.
   */
  children: PropTypes.node.isRequired,
};

export const CardBodyWithContext = withProviderContext(CardBody, 'CardBody');

export default CardBodyWithContext;
