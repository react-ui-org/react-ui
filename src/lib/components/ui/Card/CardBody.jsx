import PropTypes from 'prop-types';
import React from 'react';
import styles from './Card.scss';

const CardBody = (props) => {
  const {
    children,
  } = props;

  return (
    <div className={styles.body}>
      {children}
    </div>
  );
};

CardBody.propTypes = {
  children: PropTypes.node.isRequired,
};

export default CardBody;
