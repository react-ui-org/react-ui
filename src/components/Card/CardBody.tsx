import React from 'react';
import { withGlobalProps } from '../../providers/globalProps';
import { transferProps } from '../../helpers/transferProps';
import { CardBodyProps } from './CardBody.types';
import styles from './Card.module.scss';

export const CardBody: React.FunctionComponent<CardBodyProps> = ({
  children,
  ...restProps
}) => (
  <div
    {...transferProps(restProps)}
    className={styles.body}
  >
    {children}
  </div>
);

export const CardBodyWithGlobalProps = withGlobalProps(CardBody, 'CardBody');

export default CardBodyWithGlobalProps;
