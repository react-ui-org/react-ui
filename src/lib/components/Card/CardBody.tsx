import React from 'react';
import { withGlobalProps } from '../../provider';
import { transferProps } from '../_helpers/transferProps';
import styles from './Card.scss';
import { CardBodyProps } from './Card.types';

export const CardBody: React.FunctionComponent<CardBodyProps> = ({
  children,
  ...restProps
}) => (
  <div
    {...transferProps(restProps)}
    className={styles.body}
  >
    {children}
  </div>
);

export const CardBodyWithGlobalProps = withGlobalProps(CardBody, 'CardBody');

export default CardBodyWithGlobalProps;
