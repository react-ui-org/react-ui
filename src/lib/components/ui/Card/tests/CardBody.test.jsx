import React from 'react';
import { shallow } from 'enzyme';
import CardBody from '../CardBody';

describe('rendering', () => {
  it('renders correctly', () => {
    const tree = shallow(<CardBody><p>Card body content</p></CardBody>);

    expect(tree).toMatchSnapshot();
  });
});
