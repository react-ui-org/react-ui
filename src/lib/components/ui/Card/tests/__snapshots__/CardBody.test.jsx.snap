// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders correctly 1`] = `
<div
  className="body"
>
  <p>
    Card body content
  </p>
</div>
`;
